stdlib/Printf/src/Printf.jl

```diff
@@ -79,16 +79,18 @@ char(::Type{Val{c}}) where {c} = c
 
 # parse format string
 function Format(f::AbstractString)
-    isempty(f) && throw(ArgumentError("empty format string"))
+    isempty(f) && throw(ArgumentError("Format string must be non-empty"))
     bytes = codeunits(f)
     len = length(bytes)
     pos = 1
     b = 0x00
+
+    # skip ahead to first format specifier
     while pos <= len
         b = bytes[pos]
         pos += 1
         if b == UInt8('%')
-            pos > len && throw(ArgumentError("invalid format string: '$f'"))
+            pos > len && throw(ArgumentError("Format string is invalid - first format specifier incomplete at end of string: '$f'"))
```

Comment:
Thank you for all those suggestions! They all sound better, I'll incorporate them shortly.

Yes, the idea was to make it immediately known what threw an `ArgumentError`, without having to dive into the stacktrace. I am considering dropping the `at end of string` part, if I add the better localization of the invalid format.

I was thinking it could display like this:

```
Error: ArgumentError: Format string is invalid - first format specifier incomplete:
    'foo%'
    ----^
[..stacktrace..]
```

With the `%` in red.

Reply:
Another option which I think might be even better is to introduce a new error type `InvalidFormatStringError`.

Reply:
I think julia is generally more conservative when introducing new errors, so some input from a core dev would be welcome here.

Reply:
Oh I can't actually do the pretty printing without a new error type :/ I'll add one for the pretty printing, let's see if people like it.